### PR TITLE
Notification System

### PR DESCRIPTION
**Pull request by marcosschlup**
Added notification system based in Websocket. The payload format is Firebase friendly.

## Empty data
Captura de Tela 2020-02-25 às 18 00 18

## Read and Unread
Captura de Tela 2020-02-25 às 17 20 51

### CODE REVIEW

File: config/websocket.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  });
+
+  socket.on('markread.all', () => {
+    if (socket.userData && socket.userData._id) {
+      notificationCtrl.markReadAll(socket.userData._id);
+    }
+  });
+}
+
+function clearDisconnected() { sockets = sockets.filter(s => s.connected); }
```
Comment (sterlingrules) [inline config/websocket.js]: Your code looks good overall. My only comment is that, with lines like this, I think they should be formatted like below to improve readability and diffability — so if it changes in the future, it's clear what was changed.

```
function clearDisconnected() {
  sockets = sockets.filter(s => s.connected);
}
```

File: config/websocket.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+function askRegister(socket) {
+  socket.emit('app.register');
+}
+
+function registerSocket(socket, token) {
+  try {
+    const data = jwt.verify(token, config.jwtSecret);
+    if (!data.username) return false;
```
Comment (sterlingrules) [inline config/websocket.js]: Similar to my comment on the single line function, I think these should be in a block format, but again, your PR is fine to merge.

```
if (!data.username) {
  return false;
}
```

File: config/websocket.js (hunk 1 above)
Comment (marcosschlup) [reply config/websocket.js]: Agreed!

File: config/websocket.js (hunk 2 above)
Comment (marcosschlup) [reply config/websocket.js]: This I think the opposite. I find it easier and simpler to use a single line, single sentence. But we are practically talking about taste here

Comment (sterlingrules) [reply config/websocket.js]: Totally

**Pull request merged**